builder/amazon/common/step_security_group.go

```diff
@@ -55,8 +55,12 @@ func (s *StepSecurityGroup) Run(state multistep.StateBag) multistep.StepAction {
 	group := &ec2.CreateSecurityGroupInput{
 		GroupName:   &groupName,
 		Description: aws.String("Temporary group for Packer"),
-		VpcId:       &s.VpcId,
 	}
+
+	if s.VpcId != "" {
```

Comment:
the PR #4879 seems to fix only if a subnet is specify, in the case of using packer outside of a VPC, this option is still an empty string